eclair-core/src/main/scala/fr/acinq/eclair/db/PaymentsDb.scala

```diff
@@ -19,70 +19,131 @@ package fr.acinq.eclair.db
 import java.util.UUID
 
 import fr.acinq.bitcoin.ByteVector32
-import fr.acinq.eclair.MilliSatoshi
-import fr.acinq.eclair.payment.PaymentRequest
+import fr.acinq.bitcoin.Crypto.PublicKey
+import fr.acinq.eclair.payment._
+import fr.acinq.eclair.router.Hop
+import fr.acinq.eclair.{MilliSatoshi, ShortChannelId}
 
 trait PaymentsDb {
 
-  // creates a record for a non yet finalized outgoing payment
+  /** Create a record for a non yet finalized outgoing payment. */
   def addOutgoingPayment(outgoingPayment: OutgoingPayment)
 
-  // updates the status of the payment, if the newStatus is SUCCEEDED you must supply a preimage
-  def updateOutgoingPayment(id: UUID, newStatus: OutgoingPaymentStatus.Value, preimage: Option[ByteVector32] = None)
+  /** Update the status of the payment in case of success. */
+  def updateOutgoingPayment(paymentResult: PaymentSent)
 
+  /** Update the status of the payment in case of failure. */
+  def updateOutgoingPayment(paymentResult: PaymentFailed)
+
+  /** Get an outgoing payment attempt. */
   def getOutgoingPayment(id: UUID): Option[OutgoingPayment]
 
-  // all the outgoing payment (attempts) to pay the given paymentHash
+  /** Get all the outgoing payment attempts to pay the given paymentHash. */
   def getOutgoingPayments(paymentHash: ByteVector32): Seq[OutgoingPayment]
 
-  def listOutgoingPayments(): Seq[OutgoingPayment]
+  /** Get all the outgoing payment attempts in the given time range. */
+  def listOutgoingPayments(from: Long, to: Long): Seq[OutgoingPayment]
 
+  /** Add a new payment request (to receive a payment). */
   def addPaymentRequest(pr: PaymentRequest, preimage: ByteVector32)
 
+  /** Get the payment request for the given payment hash, if any. */
   def getPaymentRequest(paymentHash: ByteVector32): Option[PaymentRequest]
```

Comment:
All the `xxxPaymentRequest` method are very confusing and do not point out that they deal with `IncomingPayment`. You actually need to read the `SqlitePaymentDb` code to understand that they return the `PaymentRequest` for `IncomingPayment` only. 

Instead of returning a `PaymentRequest` object, these methods should return the actual `IncomingPayment` object containing the payment request, which can be unwrapped afterwards if needed.

Also, the `getIncomingPayment` and `listIncomingPayments` methods only return the settled incoming payments which is not obvious.

- change `getPaymentRequest` to:

```scala
/** Get the incoming payment (paid or not) for the given payment hash, if any. */
def getIncomingPayment(paymentHash: ByteVector32): Option[IncomingPayment]
```

- change `getIncomingPayment` to:

```scala
/** Get the paid incoming payment for the given payment hash, if any. If the payment exists but has expired or has not been paid yet, returns nothing. */
def getPaidIncomingPayment(paymentHash: ByteVector32): Option[IncomingPayment]
```

- change `getPendingPaymentRequestAndPreimage` to:

```scala
/** Get a pending incoming payment for the given payment hash, if any. If the payment exists but has been paid or has expired, returns nothing. */
def getPendingIncomingPayment(paymentHash: ByteVector32): Option[IncomingPayment]
```

- change `listPaymentRequests` , `listPendingPaymentRequests` and `listIncomingPayments` to:

```scala
/** List all incoming payments (pending, expired and fulfilled) in the given time range */
def listIncomingPayments(from: Long, to: Long): Seq[IncomingPayment]

/** List all pending incoming payments in the given time range */
def listPendingIncomingPayments(from: Long, to: Long): Seq[IncomingPayment]

/** List all paid incoming payments in the (optional) given time range */
def listPaidIncomingPayments(from: Option[Long], to: Option[Long]): Seq[IncomingPayment]
```

Reply:
Completely agree, this is a good opportunity to re-work on this interface and make it more intuitive.

Reply:
Why are the `from`/`to` fiels optional only for `listPaidIncomingPayments`?

Reply:
> Why are the `from`/`to` fields optional only for `listPaidIncomingPayments`?

My bad, the `from`/`to` fields should be optional in all 3 methods. But now I'm thinking we should not add to much stuff into this PR, and perhaps should just change the methods' name. I think at some point we will have to add more powerful filtering parameters.